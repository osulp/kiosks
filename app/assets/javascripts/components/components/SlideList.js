import React, { Component, PropTypes } from 'react';

class SlideList extends Component {

  render() {

    const { slides, setSlides } = this.props;

    const handleSubmit = (e) => {
      e.preventDefault();
    };

    const handleKeyUp = (e) => {
      console.log(`handleKeyUp, ${e.keyCode}`);
      if(e.keyCode == 13){
        if (typeof App !== 'undefined'){
          //App.room.speak(e.target.value);
        }else{
          setSlides();
        }
        e.target.value = "";
      };
    };

    return (
      <div>
        <ul>
          {slides.map((s) => {
            return <li key={`slide.${s.id}`}>{s.content}</li>;
          })
          }
        </ul>
        <form onSubmit={handleSubmit}>
          <input type="text" onKeyUp={handleKeyUp}/>
        </form>
      </div>
    );
  }
}

SlideList.propTypes = {
  slides: PropTypes.any,
};

export default SlideList;