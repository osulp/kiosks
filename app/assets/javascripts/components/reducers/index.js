import { combineReducers } from 'redux';
import slidesReducer from './slidesReducer';

const rootReducer = combineReducers({
  slides: slidesReducer
});

export default rootReducer;