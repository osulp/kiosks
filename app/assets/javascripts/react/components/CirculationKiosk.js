import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Kiosk from './presentational/Circulation/Kiosk';
import * as actions from '../actions';

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = (state) => {
  return {
    slides: state.kiosk.slides,
    url: state.kiosk.url,
    is_fetching_slides: state.touch.is_fetching_slides,
    show_nav: false,
    rooms_available_count: state.circ.rooms_available_count,
  }
};

// Map the actions to the dispatcher
export const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(Kiosk);
