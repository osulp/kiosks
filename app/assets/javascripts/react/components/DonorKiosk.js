import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Kiosk from './presentational/Donor/Kiosk';
import * as actions from '../actions';

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = (state) => {
  return {
    slides: state.kiosk.slides,
    title: state.kiosk.title,
    google_analytics: state.kiosk.google_analytics
  }
};

// Map the actions to the dispatcher
export const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(Kiosk);
