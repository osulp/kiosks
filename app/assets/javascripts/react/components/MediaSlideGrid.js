import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import MediaGrid from './presentational/Media/MediaGrid';
import * as actions from '../actions';

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = (state) => {
  return {
    slides: state.kiosk.slides,
    title: state.kiosk.title,
    is_modal_visible: state.modal.visible
  }
};

// Map the actions to the dispatcher
export const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(MediaGrid);
