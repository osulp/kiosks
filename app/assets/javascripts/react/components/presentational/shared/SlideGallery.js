import {connect} from 'react-redux';
import React, {Component, PropTypes} from 'react';
import ImageGallery from 'react-image-gallery';

class SlideGallery extends Component {
  onSlide() {
  }

  onClick(e) {
    console.log("onClick", e.target.src);
  }

  onImageError(e) {
    let error_message = `Unable to load: ${e.target.src}`;
    console.log(error_message);
    this.props.addError(error_message);
  }

  onImageLoad(e) {
    console.log(`Image Loaded:${e.target.src})`);
  }

  itemPath(item) {
    let path = '';

    switch(item.current_kiosk) {
      case 'tall':
        path = item.xtall;
        break;
      default:
        path = item.original;
        break;
    }
    return path;
  }

  _renderItem(item) {
    const onImageError = this.onImageError;
    const onImageLoad = this.onImageLoad;

    return (
      <div className='image-gallery-image'>
        <img src={this.itemPath(item)}
             alt={`Image:${item.id} : ${item.original}`}
             srcSet={item.srcSet}
             sizes={item.sizes}
             onLoad={onImageLoad.bind(this)}
             onError={onImageError.bind(this)}/>
      </div>
    )
  }

  componentDidUpdate() {
    this._imageGallery.slideToIndex(this.props.starting_slide_index);
  }

  render() {
    let slides = this.props.slides;
    // Set the slide length and if no value exists then use the minimum of 5 seconds
    let slide_length = this.props.slides[0].slide_length || 5000
    return (
      <ImageGallery ref={i => this._imageGallery = i }
                    items={slides}
                    showThumbnails={false}
                    showNav={this.props.show_nav}
                    autoPlay={true}
                    showFullscreenButton={false}
                    showPlayButton={false}
                    showBullets={true}
                    onSlide={this.onSlide}
                    onClick={this.onClick}
                    onImageError={this.onImageError}
                    onImageLoad={this.onImageLoad}
                    renderItem={this._renderItem.bind(this)}
                    startIndex={this.props.starting_slide_index}
                    slideInterval={slide_length}/>
    );
  }
}

SlideGallery.propTypes = {
  starting_slide_index: PropTypes.number.isRequired,
  slides: PropTypes.array.isRequired,
  addError: PropTypes.func.isRequired,
};

export default SlideGallery;
