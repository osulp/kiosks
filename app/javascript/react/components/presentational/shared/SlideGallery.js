import { connect } from "react-redux"
import React, { Component } from "react"
import PropTypes from "prop-types"
import ImageGallery from "react-image-gallery"

class SlideGallery extends Component {
  onSlide(index) {
    // stop all videos
    document.querySelectorAll('video').forEach(vid => {
      vid.pause()
      vid.currentTime = 0
    })

    // play if next slide is a video
    let next_item = this.props.slides[index]

    if (next_item != undefined) {
      let myNextVideo = document.querySelector(`video.video-${next_item.id}`)
      if (myNextVideo != undefined) {
        this._imageGallery.pause()
        myNextVideo.load()
        myNextVideo.play()
      } else {
        this._imageGallery.play()
      }
    }
  }

  onClick(e) {
    console.log("onClick", e.target.src)
  }

  onImageError(e) {
    let error_message = `Unable to load: ${e.target.src}`
    console.log(error_message)
    this.props.addError(error_message)
  }

  onImageLoad(e) {
    console.log(`Image Loaded:${e.target.src})`)
  }

  prevSlide(e) {
    let currentIndex = this._imageGallery.getCurrentIndex()
    let prev_index = (currentIndex - 1 + this.props.slides.length) % this.props.slides.length
    this._imageGallery.slideToIndex(prev_index)
  }

  nextSlide(e) {
    let currentIndex = this._imageGallery.getCurrentIndex()
    let next_index = (currentIndex + 1) % this.props.slides.length
    this._imageGallery.slideToIndex(next_index)
  }

  onVideoEnded(e) {
    this.nextSlide()
  }

  itemPath(item) {
    let path = ""

    switch (item.current_kiosk) {
      case "tall":
        path = item.xtall
        break
      // NEW CASE: Add in a new case for the kiosk slide
      case "interactive-touch":
        path = item.image_slide
        break
      default:
        path = item.original
        break
    }
    return path
  }

  _renderItem(item) {
    const onImageError = this.onImageError
    const onImageLoad = this.onImageLoad
    const onVideoEnded = this.onVideoEnded

    return (
      <div className="image-gallery-image">
        { 
          item.av_media == undefined ?
            <img
              src={this.itemPath(item)}
              alt={`Image:${item.id} : ${item.original}`}
              srcSet={item.srcSet}
              sizes={item.sizes}
              onLoad={onImageLoad.bind(this)}
              onError={onImageError.bind(this)}
            />
          :
            <video muted className={`video-${item.id}`} onEnded={onVideoEnded.bind(this)} preload="auto">
              <source src={item.av_media} type='video/mp4'></source>
              {item.subtitle_en.length > 0 &&
                <track
                  kind="subtitles"
                  label="English subtitles"
                  src={item.subtitle_en}
                  srcLang="en"
                  default
                />
              }
              {item.subtitle_en.length > 0 &&
                <track
                  kind="subtitles"
                  label="Spanish subtitles"
                  src={item.subtitle_es}
                  srcLang="es"
                />
              }
            </video>
        }
      </div>
    )
  }

  componentDidUpdate() {
    this._imageGallery.slideToIndex(this.props.starting_slide_index)
  }

  render() {
    let slides = this.props.slides
    // Set the slide length and if no value exists then use the minimum of 5 seconds
    let slide_length = this.props.slides[0].slide_length
    // TODO: make a new component or customize ImageGallery for AdminKiosk. 
    // autoPlay: default should be true, but on AdminKiosk, it's expected to be false.
    return (
      <ImageGallery
        ref={i => (this._imageGallery = i)}
        items={slides}
        showThumbnails={false}
        showNav={this.props.show_nav}
        autoPlay={true}
        showFullscreenButton={false}
        showPlayButton={false}
        showBullets={true}
        onSlide={this.onSlide.bind(this)}
        onClick={this.onClick}
        onImageError={this.onImageError}
        onImageLoad={this.onImageLoad}
        renderItem={this._renderItem.bind(this)}
        startIndex={this.props.starting_slide_index}
        slideInterval={slide_length}
      />
    )
  }
}

SlideGallery.propTypes = {
  starting_slide_index: PropTypes.number.isRequired,
  slides: PropTypes.array.isRequired,
  addError: PropTypes.func.isRequired
}

export default SlideGallery
