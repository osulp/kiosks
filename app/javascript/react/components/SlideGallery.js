import { bindActionCreators } from "redux"
import { connect } from "react-redux"

import SlideGallery from "./presentational/shared/SlideGallery"
import * as actions from "../actions"

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = state => {
  return {
    slides: state.kiosk.slides,
    starting_slide_index: state.kiosk.starting_slide_index
  }
}

// Map the actions to the dispatcher
export const mapDispatchToProps = dispatch => {
  return bindActionCreators(actions, dispatch)
}

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SlideGallery)
