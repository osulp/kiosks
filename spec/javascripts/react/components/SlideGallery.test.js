import { mapStateToProps, mapDispatchToProps } from '../../../../app/assets/javascripts/react/components/SlideGallery';
import * as factories from '../.factories';

describe('SlideGallery', () => {
  it('maps state to props', () => {
    expect(mapStateToProps({kiosk: { slides: factories.slides } })).toMatchSnapshot();
  });

  it('maps dispatch to props', () => {
    expect(mapDispatchToProps(jest.fn())).toMatchSnapshot();
  });
});
