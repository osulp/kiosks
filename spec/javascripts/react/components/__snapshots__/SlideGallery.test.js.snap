exports[`SlideGallery maps dispatch to props 1`] = `
Object {
  "addError": [Function],
  "fetchHours": [Function],
  "fetchSlides": [Function],
  "fetchedHours": [Function],
  "fetchedSlides": [Function],
  "fetchingHours": [Function],
  "fetchingSlides": [Function],
  "scrollToSlide": [Function],
  "setHours": [Function],
  "setKiosk": [Function],
  "setModalRootComponent": [Function],
  "setModalVisibility": [Function],
  "setSlides": [Function],
  "setTitle": [Function],
}
`;

exports[`SlideGallery maps state to props 1`] = `
Object {
  "slides": Array [
    Object {
      "caption": "Slide.caption",
      "created_at": "2000-12-31T11:59:59Z",
      "expires_at": "2020-12-31T23:59:59Z",
      "id": 1,
      "kiosk": "Kiosk.name",
      "original": "/uploads/default_slide_1.png",
      "slide_type": "SlideType.name",
      "thumbnail": "/uploads/default_slide_1.png",
      "title": "Slide.title",
      "updated_at": "2000-12-31T23:59:59Z",
      "xlarge": "/uploads/default_slide_1.png",
    },
    Object {
      "caption": "Slide.caption:2",
      "created_at": "2000-11-11T11:00:00Z",
      "expires_at": "2010-11-11T13:00:00Z",
      "id": 2,
      "kiosk": "Kiosk.name:2",
      "original": "/uploads/default_slide_2.png",
      "slide_type": "SlideType.name:2",
      "thumbnail": "/uploads/default_slide_1.png",
      "title": "Slide.title:2",
      "updated_at": "2000-11-11T13:00:00Z",
      "xlarge": "/uploads/default_slide_1.png",
    },
  ],
  "starting_slide_index": undefined,
}
`;
